### DISCUSSION TITLE
Nested vistualizers

### DISCUSSION THREAD

**Discussion by suryaashish-tyke** [TanStack/virtual#861] [Q&A] [answered]
<img src="https://github.com/user-attachments/assets/6ae92d69-4920-437b-9fbf-bbd87caaa53b" width="200px"/>
<br />

As shown in the above image 
I am having a list of items (Red-box) which in turn will have its own list of items (blue-box) and fetched when is opened.

The problem I am facing is when first open the nested list I am getting a blank list

https://github.com/user-attachments/assets/9e9f6442-1ff8-4b3f-9e2f-84de5e54e89a

**Comment by piecyk** [comment]
Each nested list needs to have the correct `scrollMargin` set, which determines where the scroll offset should start. Additionally, all the nested lists should share the same scrolling element to ensure proper rendering.

**Reply by suryaashish-tyke** [reply]
Thanks @piecyk for the reply, will try this and update you

**Comment by johnhok** [comment]
@suryaashish-tyke I saw your question here and it looks somewhat similar to what I've been trying to do. I'm assuming the blue bits are also intended to be virtualized? Did you have any luck with your implementation? I've been having issues with scroll when trying to nest virtualizers like this. Any pointers would be helpful!

Link to my thread with a code sample: https://github.com/TanStack/virtual/discussions/876

**Reply by suryaashish-tyke** [reply]
No luck @johnhok

**Reply by suryaashish-tyke** [reply, marked as answer]
Found the issue @piecyk , @johnhok 

In my case the div which holds the scrollable container, is conditionally rendered and the initial ref being `null` the contents are not getting rendered(since there is no container(ref))

![image](https://github.com/user-attachments/assets/6862f72b-0cd3-4fd1-8e2e-cdd389dfbf08)

With this change now whenever the `VirtualList` is rendered the ref will be attached and having a value.
